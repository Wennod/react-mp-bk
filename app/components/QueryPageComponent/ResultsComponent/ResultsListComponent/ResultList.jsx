import React from 'react';

import ListItem from '../ResultsListItemComponent/ResultListItem';

const ResultList = (props) => {
    if (props.data && props.data.length) {
        return (
            <div className="results">
                {props.data.map((dataItem) => {
                    return <ListItem key={dataItem.id} data={dataItem} />
                })}
            </div>
        );
    } else {
        return (
            <div className="results">
                <h2 className="results__error">No results</h2>
            </div>
        );
    }
}

export default ResultList;