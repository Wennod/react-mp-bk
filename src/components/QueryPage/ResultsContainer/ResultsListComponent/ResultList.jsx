import React from 'react';

import ListItem from './ListItemComponent/ResultListItem';

import Style from './ResultList.style.scss';

const ResultList = (props) => {
    if (props.data && props.data.length) {
        return (
            <div className="list">
                <div className="list__content">
                    {props.data.map((dataItem) => {
                        return <ListItem key={dataItem.id} data={dataItem} />
                    })}
                </div>
            </div>
        );
    } else {
        return (
            <div className="list">
                <h2 className="results__error">No results</h2>
            </div>
        );
    }
}

export default ResultList;