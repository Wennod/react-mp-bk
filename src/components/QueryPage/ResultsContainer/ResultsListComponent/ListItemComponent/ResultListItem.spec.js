import React from 'react';
import ResultList from './ResultListItem';

const __data = {
    "id": 412302,
    "title": "Gemini",
    "tagline": "",
    "vote_average": 10,
    "vote_count": 3,
    "release_date": "2018-03-30",
    "poster_path": "https: //image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg",
    "budget": 0,
    "revenue": 0,
    "genres":  ["Mystery","Thriller"],
    "runtime": 92
}

describe('ResultList', () => {
    it('should match snapshot', () => {
        const tree = mount(<ResultList data={__data} />);
        expect(tree).toMatchSnapshot();
    });
});