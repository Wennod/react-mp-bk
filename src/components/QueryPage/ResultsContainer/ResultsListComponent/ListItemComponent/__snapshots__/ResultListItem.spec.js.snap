// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList should match snapshot 1`] = `
<div
  className="movie-card"
>
  <img
    alt="Gemini poster"
    className="movie-card__image"
    src="https: //image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg"
  />
  <div
    className="movie-card__description"
  >
    <p
      className="movie-card__title"
    >
      Gemini
    </p>
    <p
      className="year"
    >
      2018
    </p>
  </div>
  <p
    className="movie-card__genre"
  >
    Mystery
  </p>
</div>
`;
